doc/async-system.md

```diff
@@ -0,0 +1,209 @@
+
+Most applications have a single `AsyncSystem` that is used throughout. It is constructed from an [ITaskProcessor](@ref CesiumAsync::ITaskProcessor) instance, which is a simple interface used to perform some work in a background thread. The simplest possible implementation looks like this:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp simplest-task-processor
+
+This implementation will work well, but it isn't very efficient because a brand new thread is created for each task. Most applications will implement this interface using a thread pool, task graph, or similar functionality that their application already contains.
+
+The `AsyncSystem` can then be created as follows:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp create-async-system
+
+`AsyncSystem` instances can be safely and efficiently stored and copied by value. This makes it easy to make them available wherever they're needed, including in lambda captures. You can think of an `AsyncSystem` instance as acting a bit like a smart pointer.
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp capture-by-value
+
+However, it is essential that the last `AsyncSystem` instance be destroyed only after everything is done using it. So a common pattern is to create and store it as a static local in an accessor function:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp async-system-singleton
+
+## Future<T> {#future}
+
+[Future<T>](@ref CesiumAsync::Future) represents an asynchronous operation that will eventually complete and produce a value of type `T`. If you're familiar with a JavaScript [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), the concept is very much the same. We commonly see `Future<T>` as the return value of a method that starts an asynchronous operation. For example, we can use [IAssetAccessor](@ref CesiumAsync::IAssetAccessor) to asynchronously download the content of a web URL:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp create-request-future
+
+This [get](@ref CesiumAsync::IAssetAccessor::get) function will start the HTTP request, but it will return quickly, without waiting for the request to complete. Initially, the `Future` is in the _Pending_ state, meaning that the asynchronous operation has not yet completed. Once it does complete, the `Future` will be in one of these two states:
+
+* _Resolved_: The operation has completed successfully and the result value is available.
+* _Rejected_: The operation failed, and an exception describing the error is available.
+
+If we want to block the current thread and wait for the request to complete, we can call [waitInMainThread](@ref CesiumAsync::Future::waitInMainThread):
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp wait-in-main-thread
+
+`waitInMainThread` blocks until the `Future` completes. If it is resolved, `waitInMainThread` returns its value. If it is rejected, `waitInMainThread` throws an exception representing the error.
+
+> [!note]
```

Comment:
I don't think you need the arrow bracket for a Doxygen `\note`. [According to the documentation](https://www.doxygen.nl/manual/commands.html#cmdpar), you should be able to write this as:
```\note
The `std::move` is needed because `Future<T>` is meant to be consumed just once. This allows for best efficiency because the value resulting from the asynchronous operation can be provided to the caller without ever making a copy of it.
```

As it is, this renders as a blockquote that's prefixed with the text `[!note]`, which I don't think is the intention.